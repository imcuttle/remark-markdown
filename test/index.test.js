/**
 * @file index
 * @author Cuttle Cong
 * @date 2018/2/24
 * @description
 */
const remark = require('remark')
const html = require('remark-html')
const stringify = require('remark-stringify')
const mark = require('../index')
const openCloseTag = require('../utils/html').openCloseTag

describe('html openTag', function () {
  it('<div></div>', () => {
    expect(openCloseTag.test('<div></div>')).toBeTruthy()
  })

  it('<div>asd</div>', () => {
    expect(openCloseTag.test('<div>sss</div>')).toBeTruthy()
  })

  it('<div a="xx"/>', () => {
    expect(openCloseTag.test('<div a="xx"/>')).toBeTruthy()
  })

  it('<0ss a="xx"/>', () => {
    expect(openCloseTag.test('<div a="xx"/>')).toBeTruthy()
  })
  it('aa<0ss a="xx"/>', () => {
    expect(openCloseTag.test('aa<div a="xx"/>')).toBeFalsy()
  })

  it('</div>', () => {
    expect(openCloseTag.test('</div>')).toBeTruthy()
  })
})

describe('main', function () {
  test('illegal block', () => {
    remark()
      .use(html)
      .use(mark)
      .process('asd<markdown>## h</markdown>', function (err, content) {
        expect(err).toBeNull()
        expect(content.contents.trim()).toBe('<p>asd<markdown>## h</markdown></p>')
      })
  })

  test('right block', () => {
    remark()
      .use(html)
      .use(mark)
      .process('<markdown>## h #x\n ## hh</markdown>', function (err, content) {
        expect(err).toBeNull()
        expect(content.contents.trim()).toBe('<markdown>\n'
                                             + '<h2>h #x</h2>\n'
                                             + '<h2>hh</h2>\n'
                                             + '</markdown>')
      })
  })

  test('right block case', () => {
    remark()
      .use(html)
      .use(mark)
      .process('<markdown>\n'
               + '## h #x\n '
               + '## hh\n'
               + '</markdown>', function (err, content) {
        expect(err).toBeNull()
        expect(content.contents.trim()).toBe('<markdown>\n'
                                             + '<h2>h #x</h2>\n'
                                             + '<h2>hh</h2>\n'
                                             + '</markdown>')
      })
  })

  test('right block case 2', () => {
    remark()
      .use(html)
      .use(mark)
      .process('<span>abc</span>\n'
               + '<div>'
               + '<span>a</span>\n'
               + '<markdown>\n'
               + '## hxx\n '
               + '## hh\n'
               + '<div>abc</div>\n'
               + '</markdown>'
               + '<span>abc</span>'
               + '</div>', function (err, content) {
        expect(err).toBeNull()
        expect(content.contents.trim()).toBe(
          '<p><span>abc</span></p>\n'
          + '<div>'
          + '<span>a</span>\n'
          + '<markdown>\n'
          + '<h2>hxx</h2>\n'
          + '<h2>hh</h2>\n'
          + '<div>abc</div>\n'
          + '</markdown>\n'
          + '<span>abc</span>'
          + '</div>'
        )
      })
  })

  test('inline html', () => {
    remark()
      .use(html)
      .use(mark)
      .process('<span>xxx</span></div>', function (err, content) {
        expect(content.contents.trim()).toBe('<p><span>xxx</span></div></p>')
      })
  })

  test('block html ', () => {

    remark()
      .use(html)
      .use(mark)
      .process('<div><span>xxx<markdown># a</markdown>hh<markdown># b</markdown></span></div>', function (err, content) {
        expect(content.contents.trim()).toBe('<div><span>xxx<markdown>\n'
                                             + '<h1>a</h1>\n'
                                             + '</markdown>\n'
                                             + 'hh'
                                             + '<markdown>\n'
                                             + '<h1>b</h1>\n'
                                             + '</markdown>\n'
                                             + '</span></div>')
      })
  })


  test('complicated', function () {
    var str = require('fs').readFileSync(__dirname + '/fixture/main.md').toString()
    remark()
      .use(html)
      .use(mark)
      .process(str, function (err, con) {
        expect(con.contents).toMatchSnapshot()
      })
  })
})
