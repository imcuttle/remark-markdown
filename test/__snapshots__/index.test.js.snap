// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main complicated 1`] = `
"<hr>
<h2>title:</h2>
<style>
  hr {
    margin: 50px 0 40px;
  }
</style>
<hr>
<h2>Get Started</h2>
<p><code>npm install picidae-theme-haier -d</code></p>
<pre><code class=\\"language-javascript\\">// picidae.config.js
module.exports = {
  theme: 'picidae-theme-haier'
}
</code></pre>
<hr>
<h2>Feature</h2>
<div class=\\"row\\">
  <div class=\\"col-sm-4\\">
    <h3 class=\\"align-center\\">
      🌎 Internationality
    </h3>
    <p>
      Haier supports internationality. <br/>
      You can write your document splited by English & Chinese. <br/>
      You can disable the feature when you just want write English or Chinese Document. <br/>
    </p>
  </div>
  
  <div class=\\"col-sm-4\\">
<markdown>
<h3>🍫 Customized</h3>
<h4>Style</h4>
<p>Supporting configuration of your style that you want.</p>
<markdown>
<h3>🍫 Customized</h3>
<h5>Header</h5>
</markdown>

<markdown>
<h2>Supporting configuration of your document that you want.</h2>
</markdown>

</markdown>

</p>
</div>
</div>
"
`;
